support/util/pki.go

```diff
@@ -0,0 +1,86 @@
+package util
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+// GenerateTestCertificate creates a test certificate with the given DNS names and IP addresses
+func GenerateTestCertificate(ctx context.Context, dnsNames []string, ipAddresses []string, duration time.Duration) ([]byte, error) {
+	var cn string
+
+	// Generate a private key
+	log, err := logr.FromContext(ctx)
+	if err != nil {
+		log.Error(err, "failed to get logger from context")
```

Comment:
why is this duplicating the error logging? by convention the base code returns formatted error messages that are bubble up to invokers
Also pretty surprising using the log which just failed?